### PR TITLE
feat(s3): Presigned urls

### PR DESCRIPTION
**Pull request by jonbjarnio**
- New S3 method that returns presigned url based on file name and type.
- Added new logging middleware that log out the incoming request.

### CODE REVIEW

File: libs/shared/modules/src/application/application.service.ts [hunk 1]
```diff
@@ -359,7 +359,7 @@ export class ApplicationService implements IApplicationService {
     applicationId: string,
     files: Array<Express.Multer.File>,
   ): Promise<ResultWrapper<S3UploadFilesResponse>> {
-    ResultWrapper.unwrap(await this.getApplication(applicationId))
+    // ResultWrapper.unwrap(await this.getApplication(applicationId))
```
Comment (osk) [inline libs/shared/modules/src/application/application.service.ts:362]:
```suggestion
```

File: libs/shared/modules/src/application/application.service.ts [hunk 2]
```diff
@@ -369,4 +369,19 @@ export class ApplicationService implements IApplicationService {
...
   }
+
+  @LogAndHandle()
+  async getPresignedUrl(
+    applicationId: string,
+    fileName: string,
+    fileType: string,
+    isOriginal: boolean,
+  ): Promise<ResultWrapper<string>> {
+    return await this.s3Service.getPresignedUrl(
```
Comment (osk) [inline libs/shared/modules/src/application/application.service.ts:380]: Þarf `await` hér?

**Review by osk — APPROVED** [review]

File: libs/shared/modules/src/application/application.service.ts (hunk 2 above)
Comment (jonbjarnio) [reply libs/shared/modules/src/application/application.service.ts:380]: Neibb svo sannarlega ekki

**Pull request merged**